app/services/user_service.py

```diff
@@ -0,0 +1,151 @@
+from app.schemas import UserResponse
+
+
+class UserService:
+    @classmethod
+    async def get_user_by_id(cls, id):
+        user = await db.fetch_one(select([User]).where(User.id == id))
+        if user is None:
+            return {
+                "message": "User not found",
+                "status_code": status.HTTP_404_NOT_FOUND,
+            }
+
+        return {
+            "message": "User found",
+            "data": UserResponse(**user),
+            "status_code": status.HTTP_200_OK,
+        }
+
+    @classmethod
+    async def get_user_by_name(cls, name):
+        user = await db.fetch_one(select([User]).where(User.username == name))
+        if user is None:
+            return {
+                "message": "User not found",
+                "status_code": status.HTTP_404_NOT_FOUND,
+            }
+        return {
+            "message": "User found",
+            "data": UserResponse(**user),
+            "status_code": status.HTTP_200_OK,
+        }
+
+    @classmethod
+    async def send_friend_request(cls, id, user):
+        friend = await cls.get_user_by_id(id)
+        if friend["status_code"] == status.HTTP_404_NOT_FOUND:
+            return friend
+
+        friend_id = friend.id
+        user_id = user.id
+        if user_id == friend_id:
+            return {
+                "message": "You can't add yourself as a friend",
+                "status_code": status.HTTP_400_BAD_REQUEST,
+            }
+
+        check_existing_request = await cls.check_exsisting_request(user_id, friend_id)
+        if check_existing_request is not None:
+            return check_existing_request
+        check_existing_request = await cls.check_exsisting_request(friend_id, user_id)
+        if check_existing_request is not None:
+            return check_existing_request
+
+        friend_exists = await db.fetch_one(
+            select([Friends])
+            .where(Friends.user_id == user_id)
+            .where(Friends.friend_id == friend_id)
+        )
+        if friend_exists is not None:
+            return {
+                "message": "User already added",
+                "status_code": status.HTTP_409_CONFLICT,
+            }
+
+        await db.execute(
+            insert(FriendRequests).values(user_id=user_id, friend_id=friend_id)
+        )
+        return {
+            "message": "User added",
+            "status_code": status.HTTP_201_CREATED,
+        }
+
+    async def check_exsisting_request(self, user_id, friend_id):
+        existing_request = await db.fetch_one(
+            select([FriendRequests])
+            .where(FriendRequests.user_id == user_id)
+            .where(FriendRequests.friend_id == friend_id)
+        )
+        if existing_request is not None:
+            return {
+                "message": "You have already sent this user a friend request",
+                "status_code": status.HTTP_400_BAD_REQUEST,
+            }
+        return None
```

Comment:
Do not use None, set the return type as `op_status: dict, resp: dict` If record exists op_status would be false, else true, in case of false send `{
                "message": "You have already sent this user a friend request",
                "status_code": status.HTTP_400_BAD_REQUEST,
            }` as resp: dict and in case of true send `{"message": "No existing request found"}`